packages/edit-site/src/components/sidebar-navigation-screen-template-part/template-part-navigation-menu.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { __experimentalHeading as Heading } from '@wordpress/components';
+import { useEntityProp } from '@wordpress/core-data';
+/**
+ * Internal dependencies
+ */
+import NavigationMenuEditor from '../sidebar-navigation-screen-navigation-menu/navigation-menu-editor';
+
+export default function TemplatePartNavigationMenu( { id } ) {
+	const [ title ] = useEntityProp( 'postType', 'wp_navigation', 'title', id );
+
+	if ( ! id ) return null;
+
+	return (
+		<>
+			<Heading
+				className="edit-site-sidebar-navigation-screen-template-part-navigation-menu__title"
+				size="12"
+				upperCase={ true }
+			>
+				{ title || __( 'Navigation Menu' ) }
```

Comment:
I think in other places where a navigation doesn't have a title we output `(no title)`. See https://github.com/WordPress/gutenberg/blob/12a01d58762215def030ab7f262a68210d1f42cc/packages/block-library/src/navigation/edit/navigation-menu-selector.js#L25

Reply:
Actually I misunderstood what this does, we just need to do this:
```suggestion
				{ title || __( 'Navigation' ) }
```